src/cascadia/QueryExtension/ExtensionPalette.cpp

```diff
@@ -286,12 +288,14 @@ namespace winrt::Microsoft::Terminal::Query::Extension::implementation
         {
             auto suggestion = winrt::to_string(selectedItemAsChatMessage.MessageContent());
 
-            // the AI sometimes sends code blocks with newlines in them
-            // sendInput doesn't work with single new lines, so we replace them with \r
+            // the AI sometimes sends multiline code blocks
+            // we don't want to run any of those commands when the chat item is clicked,
+            // so we replace newlines with the appropriate delimiter
             size_t pos = 0;
             while ((pos = suggestion.find("\n", pos)) != std::string::npos)
             {
-                suggestion.replace(pos, 1, "\r");
+                const auto delimiter = _ActiveCommandline == cmdExe ? cmdCommandDelimiter : commandDelimiter;
+                suggestion.replace(pos, 1, delimiter);
```

Comment:
If you intention is to replace a single character with a single character, I believe the most straight forward approach would be to just index into the string:
```cpp
const auto delimiter = _ActiveCommandline == L"cmd.exe" ? L';' : L';';
suggestion.at(pos) = delimiter;
```